### DISCUSSION TITLE
Build issue

### DISCUSSION THREAD

**Discussion by oleksandrmelnychenko** [microsoft/msquic#1733] [Q&A]
Can somebody help me with an error, no ideas. In CMakeError.log I have got 

  CheckIncludeFile.c

  cl /c /Zi /W1 /WX- /diagnostics:column /Od /Ob0 /D WIN32 /D _WINDOWS /D "CMAKE_INTDIR=\"Debug\"" /D _MBCS /Gm- /RTC1 /MTd /GS /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /Fo"cmTC_1eff8.dir\Debug\\" /Fd"cmTC_1eff8.dir\Debug\vc142.pdb" /external:env:EXTERNAL_INCLUDE /external:W1 /Gd /TC /errorReport:queue C:\Users\oleks\Desktop\Build\CMakeFiles\CMakeTmp\CheckIncludeFile.c

C:\Users\oleks\Desktop\Build\CMakeFiles\CMakeTmp\CheckIncludeFile.c(1,10): fatal error C1083: Cannot open include file: 'pthread.h':

![image](https://user-images.githubusercontent.com/7407215/122394460-ae2dd400-cf7e-11eb-8e66-93aa16a891c6.png)

**Comment by ThadHouse** [comment]
What version of cmake and visual studio do you have installed?

**Reply by oleksandrmelnychenko** [reply]
cmake - 3.20.1
vs - 2019 community

**Reply by ThadHouse** [reply]
Ok. And you ran prepare-machine.ps1? Is there any more errors you can send? The full cmake log might be helpful.

**Reply by oleksandrmelnychenko** [reply]
![image](https://user-images.githubusercontent.com/7407215/122411626-927dfa00-cf8d-11eb-9595-e89a74dd6f1e.png)

**Reply by oleksandrmelnychenko** [reply]
the next command .\build.ps1 , shows the error with $process

**Reply by oleksandrmelnychenko** [reply]
![image](https://user-images.githubusercontent.com/7407215/122411966-db35b300-cf8d-11eb-96b6-57f6c2f74d6a.png)

**Reply by ThadHouse** [reply]
There should be more logs somewhere. Just to double check, cmake is in path, so you can run cmake manually through the command line? And can you also try restarting your terminal?

**Reply by oleksandrmelnychenko** [reply]
Determining if the include file pthread.h exists failed with the following output:
Change Dir: C:/Users/oleks/Desktop/Build/CMakeFiles/CMakeTmp

Run Build Command(s):C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/MSBuild/Current/Bin/MSBuild.exe cmTC_bb809.vcxproj /p:Configuration=Debug /p:Platform=x64 /p:VisualStudioVersion=16.0 /v:m && Microsoft (R) Build Engine version 16.10.2+857e5a733 for .NET Framework

Copyright (C) Microsoft Corporation. All rights reserved.



  Microsoft (R) C/C++ Optimizing Compiler Version 19.29.30038.1 for x64

  Copyright (C) Microsoft Corporation.  All rights reserved.

  CheckIncludeFile.c

  cl /c /Zi /W1 /WX- /diagnostics:column /Od /Ob0 /D WIN32 /D _WINDOWS /D "CMAKE_INTDIR=\"Debug\"" /D _MBCS /Gm- /RTC1 /MTd /GS /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /Fo"cmTC_bb809.dir\Debug\\" /Fd"cmTC_bb809.dir\Debug\vc142.pdb" /external:env:EXTERNAL_INCLUDE /external:W1 /Gd /TC /errorReport:queue C:\Users\oleks\Desktop\Build\CMakeFiles\CMakeTmp\CheckIncludeFile.c

C:\Users\oleks\Desktop\Build\CMakeFiles\CMakeTmp\CheckIncludeFile.c(1,10): fatal error C1083: Cannot open include file: 'pthread.h': No such file or directory [C:\Users\oleks\Desktop\Build\CMakeFiles\CMakeTmp\cmTC_bb809.vcxproj]

**Reply by oleksandrmelnychenko** [reply]
UI Cmake compilers:
vs - x64

**Reply by ThadHouse** [reply]
I wouldn't try building through the UI. I wouldn't be surprised if that doesn't work on windows. The only supported build method on windows is the powershell scripts.

**Reply by ThadHouse** [reply]
That pthread.h thing is not an actual error, that's just how cmake works. There's got to be something else.

**Reply by oleksandrmelnychenko** [reply]
I have tried both methods, but oka

**Reply by ThadHouse** [reply]
Does a debug build work?

**Reply by oleksandrmelnychenko** [reply]
the reason was: cmake was not in env variables ( sry

**Reply by oleksandrmelnychenko** [reply]
![image](https://user-images.githubusercontent.com/7407215/122414686-002b2580-cf90-11eb-8a70-849cda7ae28a.png)
does it okay ?

**Reply by ThadHouse** [reply]
Hmm. That should not be erroring like that. I wonder if somehow newer windows sdks have issues. I'll take a look later and see if I can reproduce that.

**Reply by nibanks** [reply]
What SDK versions do you have installed? This is what my Visual Studio Installer for VS Community 2019 shows:

![image](https://user-images.githubusercontent.com/20663557/122425969-ba953b00-cf44-11eb-8c80-7dcd9b06b6ea.png)

**Reply by oleksandrmelnychenko** [reply]
![image](https://user-images.githubusercontent.com/7407215/122426448-f0fca580-cf98-11eb-9cfd-ac18f803e65a.png)
Yes, that was an issue due to Windows SDK

**Reply by nibanks** [reply]
Great. Do you know what version you had before, that didn't work? I'm wondering if we can add some additional requirement checks to make sure the right minimum version is installed and give a more helpful error when it's not. @thadhouse what do you think?

**Reply by oleksandrmelnychenko** [reply]
Amm yes, the MORE than 10.0.19, I think 10.0.2....

**Reply by oleksandrmelnychenko** [reply]
Default SDK msi installer is OK

**Reply by oleksandrmelnychenko** [reply]
Build folder should be BUILD via CMake yes?

**Reply by nibanks** [reply]
I'd recommend taking a look at our build.ps1 [CMake-Generate](https://github.com/microsoft/msquic/blob/main/scripts/build.ps1#L249) and [CMake-Build](https://github.com/microsoft/msquic/blob/main/scripts/build.ps1#L343) functions for what we do normally.

**Reply by oleksandrmelnychenko** [reply]
![image](https://user-images.githubusercontent.com/7407215/122430236-1a6b0080-cf9c-11eb-9087-94b2c28f5750.png)

Include folder have 4 headers and lib have  1 dll and 1 lib files, 

it's correct yes?

**Reply by ThadHouse** [reply]
If you're using the cmake install functionality yes that is correct. That is all that is needed to use the library in another project.

**Reply by oleksandrmelnychenko** [reply]
Hope it can be used in QT on android apps, but need to do a Linux build

**Reply by oleksandrmelnychenko** [reply]
thx
